sdk/DTO/AttributeDefinition.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+using System.Text.Json.Serialization;
+
+namespace EAVFW.Extensions.Manifest.SDK
+{
+    public class AttributeConverter : JsonConverter<AttributeDefinitionBase>
+    {
+        public override AttributeDefinitionBase Read(
+            ref Utf8JsonReader reader,
+            Type typeToConvert,
+            JsonSerializerOptions options)
+        {
+            switch (reader.TokenType)
+            {
+                case JsonTokenType.String:
+                    return new AttributeStringDefinition { Value = reader.GetString() ?? "" };
+                case JsonTokenType.StartObject:
+                    var node = JsonNode.Parse(ref reader);
+                    var t = node.Deserialize<AttributeObjectDefinition>();
+                    return t;
+                default:
+                    throw new Exception($"{reader.TokenType} is not supported as AttributeDefinition");
+            }
+        }
+
+        public override void Write(Utf8JsonWriter writer, AttributeDefinitionBase value, JsonSerializerOptions options)
+        {
+            var jsonString = value switch
+            {
+                AttributeObjectDefinition attributeObjectDefinition =>
+                    JsonSerializer.Serialize(attributeObjectDefinition),
+                AttributeStringDefinition attributeStringDefinition =>
+                    JsonSerializer.Serialize(attributeStringDefinition),
+                _ => throw new ArgumentException($"{value.GetType()} is not supported")
+            };
+
+            var jsonNode = JsonNode.Parse(jsonString);
+            jsonNode?.WriteTo(writer, options);
+        }
+    }
+
+    [JsonConverter(typeof(AttributeConverter))]
+    public abstract class AttributeDefinitionBase
+    {
+    }
+
+    public class AttributeStringDefinition : AttributeDefinitionBase
```

Comment:
What case do we have where attribute is not an object?

Reply:
```json
"attributes": {
  "[merge()]": "[variables('audit')]",
  "Title": {
    "isPrimaryField": true
  },
```

Then the merge property is a string.


I build the first iteration on the non enriched version :)

Reply:
And the moduleSource and moduleLocation is a part of attributes if they are merged from a variable